independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/BootstrapAppModelResolver.java

```diff
@@ -134,68 +135,15 @@ public AppModel resolveManagedModel(AppArtifact appArtifact, List<AppDependency>
-        final VersionRangeResult rangeResult = resolveVersionRangeResult(appArtifact, appArtifact.getVersion(), false, upToVersion, inclusive);
-        final List<Version> resolvedVersions = rangeResult.getVersions();
-        final List<String> versions = new ArrayList<>(resolvedVersions.size());
-        for (Version v : resolvedVersions) {
-            versions.add(v.toString());
-        }
-        return versions;
-    }
-
-    @Override
-    public String getNextVersion(AppArtifact appArtifact, String fromVersion, boolean fromVersionIncluded, String upToVersion, boolean upToVersionInclusive) throws AppModelResolverException {
-        final VersionRangeResult rangeResult = resolveVersionRangeResult(appArtifact, fromVersion, fromVersionIncluded, upToVersion, upToVersionInclusive);
-        final List<Version> versions = rangeResult.getVersions();
-        if(versions.isEmpty()) {
-            return null;
-        }
-        Version next = versions.get(0);
-        for(int i = 1; i < versions.size(); ++i) {
-            final Version candidate = versions.get(i);
-            if(next.compareTo(candidate) > 0) {
-                next = candidate;
-            }
-        }
-        return next.toString();
-    }
-
-    @Override
-    public String getLatestVersion(AppArtifact appArtifact, String upToVersion, boolean inclusive) throws AppModelResolverException {
-        final VersionRangeResult rangeResult = resolveVersionRangeResult(appArtifact, appArtifact.getVersion(), false, upToVersion, inclusive);
-        final List<Version> versions = rangeResult.getVersions();
-        if(versions.isEmpty()) {
-            return appArtifact.getVersion();
-        }
-        Version latest = versions.get(0);
-        for(int i = 1; i < versions.size(); ++i) {
-            final Version candidate = versions.get(i);
-            if(latest.compareTo(candidate) < 0) {
-                latest = candidate;
-            }
-        }
-        return latest.toString();
-    }
-
-    public List<RemoteRepository> resolveArtifactRepos(AppArtifact appArtifact) throws AppModelResolverException {
-        return mvn.resolveDescriptor(toAetherArtifact(appArtifact)).getRepositories();
-    }
-
-    public void install(AppArtifact appArtifact, Path localPath) throws AppModelResolverException {
-        mvn.install(new DefaultArtifact(appArtifact.getGroupId(), appArtifact.getArtifactId(), appArtifact.getClassifier(),
-                appArtifact.getType(), appArtifact.getVersion(), Collections.emptyMap(), localPath.toFile()));
-    }
 
-    private AppModel injectDeploymentDependencies(AppArtifact appArtifact, DependencyNode root, List<Dependency> managedDeps) throws AppModelResolverException {
+        DependencyNode resolvedDeps = mvn.resolveManagedDependencies(toAetherArtifact(appArtifact),
+                directMvnDeps, managedDeps, managedRepos, devmode ? new String[] { "test" } : new String[0]).getRoot();
```

Comment:
Sorry to bother you with my dumb questions but trying to understand what's going on to make a real review :).

Why exactly do we exclude the test scope when in dev mode and not in the other modes?